### ISSUE TITLE
request: function to split RDI adp (ADCP) files

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1098]
This came up in the discussion of #1093 -- now that we have C code that can calculate a vector of pointers to ensemble chunks within an RDI adp file, we can create a function that will split such files into smaller subfiles. That would be convenient in the short term, until we get `read.adp.rdi()` to handle large files directly (which may be weeks away, given the priority and other commitments at this time of year).

A question is the name of the function, and its arguments. I'll type some code below to see if it "looks right". Maybe, when CR is back online next week, he will have some other ideas. (I know he has written code to split by the two-byte sequence that starts ensembles, but there is a 1/256^2 probability of this sequence occurring at a random spot.)

``` R
 # split into 5 parts, naming the a.000_001, a.000_0.002 etc
splitRdiFile("a.000", length.out=5)

# a.000_001 has profiles 1 to 99, a.000_002 is profiles 100 to 199 etc
splitRdiFile("a.000", from=1, to=1000, by=100)
```

In other words, I'm suggesting a syntax like that of `seq()`, because I think users will be familiar with that. We could also have an argument for the names of output files.  (I don't know, for example, whether RDI software has rules on filenames -- we certainly don't want to violate those.)

**Comment by richardsc** [comment]
> Maybe, when CR is back online next week, he will have some other ideas.

Yes, this could be handy, especially if large files are going to be the norm (I don't really know if my 5.7 GB file is an exception or the rule). Let me have a think about it a bit.

>  (I know he has written code to split by the two-byte sequence that starts ensembles, but there is a 1/256^2 probability of this sequence occurring at a random spot.)

Actually, what I did was a little more clever. I parsed the first two ensembles to read the field that says how many bytes are in each (not including the checksum, and remembering that for some files the first ensemble is bigger than all the others), and then split the files into sizes based on a number of bytes calculated using that value. For my 5.7 GB file it worked perfectly, but that's only one test. One thing I'm not doing (which I'm guessing some of the new `ldc_rdi.c` code does) is verify the checksum of the ensembles.

I don't think this should be too high on the priority list though. The immediate issue that started me thinking about it has been solved (my files are already split into readable chunks), so this will only become an issue again when (someone else?) has a large RDI file.

**Comment by richardsc** [comment]
I think this issue should be closed, as it has gone stale (e.g. we have no files for which it matters at this point), and there actually is a solution provided by software available from the manufacturer. Of course, that software is windoze-only, but if the need to split large files were to occur again I think the approach I took above (based on bash CLI tools) should suffice.

A better plan, I think, would be implement #1093 rather than this approach, because that is more easily accomplished in R/C and would be in line with other recent changes that were made to the rdi-reading code in the past couple months.

Thoughts @dankelley ?

**Comment by dankelley** [comment]
I agree; 1093 is more important. Closing now.
